### ISSUE TITLE
IsADirectoryError when accessing folder with a static route

### ISSUE THREAD

**Issue by SteffenDE** [aio-libs/aiohttp#921]
When accessing a URL handled by a static route, which is a directory, the server responds with HTTP 200 AND HTTP 500 in the response body instead of just responding by e.g. Forbidden.

```
$ http http://localhost:5000/web/static/
HTTP/1.1 200 OK
CONTENT-LENGTH: 204
CONTENT-TYPE: application/octet-stream
DATE: Fri, 10 Jun 2016 17:27:48 GMT
LAST-MODIFIED: Wed, 08 Jun 2016 14:01:43 GMT
SERVER: Python/3.5 aiohttp/0.21.6

HTTP/1.1 500 Internal Server Error
CONTENT-TYPE: text/html; charset=utf-8
CONTENT-LENGTH: 170
CONNECTION: close
DATE: Fri, 10 Jun 2016 17:27:48 GMT
SERVER: Python/3.5 aiohttp/0.21.6


<html>
  <hea

```

The route is added by calling

``` python
app.router.add_static("/web/static", "web/static")
```

**Comment by asvetlov** [comment]
Please make sure `web/static` both is a directory and web server process has appropriate rights.
As side note better to use absolute path to directory, it helps to avoid weird problems.

**Comment by SteffenDE** [comment]
`web/static` is a directory and the web server has appropriate rights. Accessing files inside of the directory works fine.
I realized that I forgot to include the traceback:

```
======== Running on http://127.0.0.1:8888/ ========
(Press CTRL+C to quit)
Error handling request
Traceback (most recent call last):
  File "/usr/local/lib/python3.5/site-packages/aiohttp/server.py", line 266, in start
    yield from self.handle_request(message, payload)
  File "/usr/local/lib/python3.5/site-packages/aiohttp/web.py", line 90, in handle_request
    resp = yield from handler(request)
  File "/usr/local/lib/python3.5/site-packages/aiohttp/web_urldispatcher.py", line 595, in handle
    with filepath.open('rb') as f:
  File "/usr/local/Cellar/python3/3.5.1/Frameworks/Python.framework/Versions/3.5/lib/python3.5/pathlib.py", line 1136, in open
    opener=self._opener)
IsADirectoryError: [Errno 21] Is a directory: '/Users/Steffen/Desktop/static'
```

Using an absolute path does not solve the issue.

``` python
import asyncio
from aiohttp import web

app = web.Application()
app.router.add_static("/static", "/Users/Steffen/Desktop/static")

web.run_app(app, port=8888)
```

```
$ http http://127.0.0.1:8888/static/
HTTP/1.1 200 OK
CONTENT-LENGTH: 68
CONTENT-TYPE: application/octet-stream
DATE: Sat, 11 Jun 2016 20:17:17 GMT
LAST-MODIFIED: Sat, 11 Jun 2016 20:16:51 GMT
SERVER: Python/3.5 aiohttp/0.21.6

HTTP/1.1 500 Internal Server Error
CONTENT-TYPE: text/html; charset

```

**Comment by asvetlov** [comment]
Gotcha!
aiohttp doesn't support directory indexing yet.
Please point in a browser a path to concrete file like `http://localhost/static/image.jpg`

**Comment by SteffenDE** [comment]
I think directory indexing is not needed but wouldn't it be better if the server would respond with a correct HTTP response? Sending 200 and then the headers and half a body of a 500 error does not seem right to me.

**Comment by asvetlov** [comment]
Agreed.
Let's keep the issue open until I'll find a time to fix it.
Indexing may be optional (off by default).

**Comment by trimailov** [comment]
Tasks to close the issue (discussed with @asvetlov):

If folder is accessed:
- [ ] By default throw HTTP 403 - Forbidden
- [ ] If folder indexing is enabled - list the directory
  - Probably extra methods are needed in `web_urldispatcher.py:StaticRoute` class.

**Comment by asvetlov** [comment]
Fixed by #1094
